proto/config/netinst.proto

```diff
@@ -310,4 +310,21 @@ message NetworkInstanceConfig {
   // Depending on the implementation, it may also introduce additional packet processing
   // overhead.
   bool disable_flowlog = 50;
+
+  // VLAN access ports configured for a switch network instance.
+  // For other types of network instances, this option is ignored.
+  // This setting applies to physical network ports attached to the network instance.
+  // VLAN configuration for application interfaces is applied separately via AppInstanceConfig
+  // (see NetworkAdapter.access_vlan_id).
+  repeated VlanAccessPort vlan_access_ports = 51;
 }
+
+// VLAN Access Port config applied to physical port(s) attached to a Switch Network Instance.
+message VlanAccessPort {
+  // VLAN ID from the range <1,4094>
+  uint32 vlan_id = 1;
+  // Physical network adapter used as the access port for the given VLAN.
+  // Either a single NI port referenced by its name (SystemAdapter.Name, aka logical label)
+  // or an adapter shared-label matching zero or more NI ports.
```

Comment:
Does it make sense even when it matches zero ports? Is this so it will wait for matching ports to appear later?

Reply:
This is just a generic description for shared labels.
Config that never matches any port is indeed pointless. However, there can be a case where temporarily no port is matched but later a new port appears (that was previously e.g. assigned to an app) or user adds shared label to an existing port. Zedrouter will then add the port into the switch NI and apply the VLAN config.

Reply:
ok